src/extensions/extension-discovery/is-compatible-extension/is-compatible-extension.ts

```diff
@@ -9,44 +9,18 @@ interface Dependencies {
   appSemVer: SemVer;
 }
 
-export const isCompatibleExtension = ({
-  appSemVer,
-}: Dependencies): ((manifest: LensExtensionManifest) => boolean) => {
-  const { major, minor, patch, prerelease: oldPrelease } = appSemVer;
-  let prerelease = "";
-
-  if (oldPrelease.length > 0) {
-    const [first] = oldPrelease;
-
-    if (first === "alpha" || first === "beta" || first === "rc") {
-      /**
-       * Strip the build IDs and "latest" prerelease tag as that is not really
-       * a part of API version
-       */
-      prerelease = `-${oldPrelease.slice(0, 2).join(".")}`;
-    }
-  }
-
-  /**
-   * We unfortunately have to format as string because the constructor only
-   * takes an instance or a string.
-   */
-  const strippedVersion = new SemVer(
-    `${major}.${minor}.${patch}${prerelease}`,
-    { includePrerelease: true },
-  );
+export const isCompatibleExtension = (deps: Dependencies): ((manifest: LensExtensionManifest) => boolean) => {
+  const { major: appVersion } = deps.appSemVer;
 
   return (manifest: LensExtensionManifest): boolean => {
-    if (manifest.engines?.lens) {
-      /**
-       * include Lens's prerelease tag in the matching so the extension's
-       * compatibility is not limited by it
-       */
-      return semver.satisfies(strippedVersion, manifest.engines.lens, {
-        includePrerelease: true,
+    if (manifest?.engines.lens) {
+      const { major: extMajor, minor: extMinor } = new SemVer(manifest.engines.lens, {
+        loose: true,
       });
+
+      return semver.gte(`${extMajor}.${extMinor}`, `${appVersion}.0`);
     }
 
-    return false;
+    return false; // not compatible by default
```

Comment:
Btw, if this is `false` by default should we mark `LensExtensionManifest` with required `engines.lens` field? Or how extensions know that and not ignored during compatibility check?

Reply:
We certainly should make it required elsewhere. In our extension development documentation (https://api-docs.k8slens.dev/v5.4.6/extensions/get-started/anatomy/#extension-manifest) it doesn't imply that it is optional (though we really should update our docs)